plugins/modules/k8s_taint.py

```diff
@@ -0,0 +1,275 @@
+            value:
+                description:
+                    - The taint value corresponding to the taint key.
+                type: str
+            effect:
+                description:
+                    - The effect of the taint on Pods that do not tolerate the taint.
+                    - Required when I(state=present).
+                type: str
+                choices: [ NoSchedule, NoExecute, PreferNoSchedule ]
+    overwrite:
+        description:
+            - If true, allow taints to be overwritten, otherwise reject taint updates that overwrite existing taints.
+        required: false
+        default: false
+        type: bool
+requirements:
+  - python >= 3.6
+  - kubernetes >= 12.0.0
+"""
+
+EXAMPLES = r"""
+- name: Taint node "foo"
+  kubernetes.core.k8s_taint:
+    state: present
+    name: foo
+    taints:
+        - effect: NoExecute
+          key: "key1"
+
+- name: Taint node "foo"
+  kubernetes.core.k8s_taint:
+    state: present
+    name: foo
+    taints:
+        - effect: NoExecute
+          key: "key1"
+          value: "value1"
+        - effect: NoSchedule
+          key: "key1"
+          value: "value1"
+
+- name: Remove taint from "foo".
+  kubernetes.core.k8s_taint:
+    state: absent
+    name: foo
+    taints:
+        - effect: NoExecute
+          key: "key1"
+          value: "value1"
+"""
+
+RETURN = r"""
+result:
+    description:
+        - The tainted Node object.
+    returned: success
+    type: complex
+    contains:
+        api_version:
+            description: The versioned schema of this representation of an object.
+            returned: success
+            type: str
+        kind:
+            description: Represents the REST resource this object represents.
+            returned: success
+            type: str
+        metadata:
+            description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+            returned: success
+            type: complex
+        spec:
+            description: Specific attributes of the object. Will vary based on the I(api_version) and I(kind).
+            returned: success
+            type: complex
+        status:
+            description: Current status details for the object.
+            returned: success
+            type: complex
+"""
+
+import copy
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+from ansible_collections.kubernetes.core.plugins.module_utils.common import (
+    K8sAnsibleMixin,
+    get_api_client,
+)
+from ansible_collections.kubernetes.core.plugins.module_utils.args_common import (
+    AUTH_ARG_SPEC,
+)
+
+try:
+    from kubernetes.client.api import core_v1_api
+    from kubernetes.dynamic.exceptions import ApiException
+except ImportError:
+    # ImportError are managed by the common module already.
+    pass
+
+
+def _equal_dicts(a, b):
+    return all([ a[x] == b[x] for x in ('effect', 'key') ])
```

Comment:
I don't think this change is going to work. You at least need to also look at `value`. In addition, the dictionaries being compared may not have an `effect`. For example, it should be possible to specify the following:
```
state: absent
taints:
  - key: foo
```
This should remove any taint with the key `foo`.

Reply:
So I'm wrong about the value, as that does not seem to matter. We do still need to support deleting by keys. I feel like something like this would work:
```python
def clean(d):
    return {k: v for k, v in d.items() if k in ("key", "effect")}


def within(a, b):
    return clean(a).items() <= clean(b).items()


def difference(a, b):
    diff = []
    for x in b:
        for y in a:
            if not within(x, y):
                diff.append(y)
    return diff
```

Reply:
Yes, we don't need the value for deleting, but only for updating. I'm actually finishing the tests since I added more and I'll push the changed in a while. Thanks.